localstack/packages/core.py

```diff
@@ -173,24 +188,69 @@ def _get_install_marker_path(self, install_dir: str) -> str:
         binary_name = os.path.basename(url)
         return os.path.join(install_dir, binary_name)
 
-    def get_executable_path(self) -> str | None:
+    def _install(self, target: InstallTarget) -> None:
+        target_directory = self._get_install_dir(target)
+        mkdir(target_directory)
+        download_url = self._get_download_url()
+        target_path = self._get_install_marker_path(target_directory)
+        download(download_url, target_path)
+
+
+class ArchiveDownloadAndExtractInstaller(ExecutableInstaller):
```

Comment:
I think it's self-explanatory that the archive will also be extracted.
Therefore I would suggest naming this class `ArchiveDownloadInstaller`.

Reply:
I was hesitant to choose that name because a jar is technically also an archive (which you don't extract). Change the name nonetheless?

Reply:
Good point, but yes, I would change the name nonetheless.